include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -3546,61 +3720,131 @@ MGTwoLevelTransferNonNested<dim, LinearAlgebra::distributed::Vector<Number>>::
     this->vec_coarse.reinit(this->partitioner_coarse);
   }
   {
+    // TODO: This adds to many dofs and therefore the IndexSet has to be created
+    // manually.
+    IndexSet locally_relevant_dofs;
+    if (!this->fine_element_is_continuous &&
+        !dof_handler_fine.get_fe().degree == 0)
+      DoFTools::extract_locally_relevant_dofs(dof_handler_fine,
+                                              locally_relevant_dofs);
+
     this->partitioner_fine.reset(
       new Utilities::MPI::Partitioner(dof_handler_fine.locally_owned_dofs(),
+                                      locally_relevant_dofs,
```

Comment:
This should probably moved to the end of the function!?

Reply:
I think it should be right before `rpe.reinit()` but in any case before `constraint_info.read_dof_indices()`, which needs the coarse partitioner. Or are you thinking about only moving the fine partitioner and keeping the coarse partitioner at the top of the function?

Reply:
I think both approaches would be fine.

Reply:
[be632e4](https://github.com/dealii/dealii/pull/15209/commits/be632e43106c04b28133d2139898608f01bc96dd)